src/lib/cardiac_mechanics/BaseSolverNeoHooke.hpp

```diff
@@ -0,0 +1,370 @@
+      reduction_increment_strat = other.reduction_increment_strat;
+    }
+    /**
+     * @brief Retrieve the pressure value
+     * @return The configured pressure value
+     */
+    Scalar value() const {
+      return static_cast<Scalar>(static_cast<double>(pressure) *
+                                 reduction_factor);
+    }
+    /**
+     * @brief Evaluate the pressure at a given point
+     * @param p The evaluation point
+     * @param component The component id
+     */
+    virtual Scalar value(const Point<dim> & /*p*/,
+                         const unsigned int /*component*/ = 0) const override {
+      return static_cast<Scalar>(static_cast<double>(pressure) *
+                                 reduction_factor);
+    }
+    /**
+     * @brief Increment the reduction factor
+     */
+    void increment_reduction_factor() {
+      reduction_factor += reduction_increment_strat;
+      reduction_factor = std::min(reduction_factor, 1.0);
+    }
+    /**
+     * @brief Get the reduction factor
+     * @return The current reduction factor
+     */
+    double get_reduction_factor() const { return reduction_factor; }
+
+  protected:
+    /**
+     * @brief Pressure value in Pa
+     */
+    Scalar pressure;
+    /**
+     * @brief A reduction factor to be applied to the set pressure value
+     */
+    double reduction_factor;
+    /**
+     * The pressure reduction factor increment value. Refer to the reference
+     * paper for the employed modfied Newton iteration
+     */
+    double reduction_increment_strat;
+  };
+  /**
+   * @brief Determine if a given faces is at a Neumann boundary
+   * @param face The face values
+   * @return The requested query
+   */
+  bool is_face_at_neumann_boundary(const unsigned face) {
+    return neumann_boundary_faces.find(face) != neumann_boundary_faces.end();
+  }
+  /**
+   * @brief Constructor
+   * @param problem_name_ The problem name
+   */
+  BaseSolverNeoHooke(const std::string &problem_name_)
+      : mpi_size(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD)),
+        mpi_rank(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)),
+        pcout(std::cout, mpi_rank == 0), mesh(MPI_COMM_WORLD),
+        problem_name(problem_name_) {
+    // Trillinos library supports only doubles for now.
+    static_assert(std::is_same_v<Scalar, double>, "Type not supported");
+  }
+  /**
+   * @brief Virtual destructor for abstract class
+   */
+  virtual ~BaseSolverNeoHooke() {}
+
+  virtual void setup();
+
+  virtual void solve_newton();
+
+  virtual void output() const;
+
+protected:
+  /**
+   * @brief Initialise boundaries tag. This must be implemented by the derived
+   * class
+   */
+  virtual void initialise_boundaries_tag() = 0;
+  /**
+   * @brief Initialise parameters handler. This must be implemented by the
+   * derived class
+   * @param file_ The input parameters file
+   */
+  virtual void initialize_param_handler(const std::string &file_) = 0;
```

Comment:
I agree that this could be made cleaner and better, the long story short for this choice is that we needed to align some std::cout, hence only a visual accommodation, probability not worth breaking good programming paradigms for this cause.